### PR TITLE
Added support for Unicode Combining Diacritical Marks in FlxBitmapText

### PR DESCRIPTION
**Pull request by SeiferTim**
For applyng Underlines, Umlauts, and other combination characters to your Bitmap Text.
More information: https://en.wikipedia.org/wiki/Combining_Diacritical_Marks

Example Usage: add an underline character to your BitmapFont and give it the Id of 818 (Decimal for U+0332), make sure it is as wide as your widest xadvance + letterspacing for your font. Set it's xadvance = 0

In your code, set your `FlxBitmapText.text = "This M\u0332 is Underlined!";` and it should show up with the letter M underlined!

### CODE REVIEW

**Comment by Gama11** [comment]
That diff primarily seems to consist of unrelated formatting changes...

File: flixel/text/FlxBitmapText.hx [hunk 1]
```diff
@@ -1182,6 +1197,8 @@ class FlxBitmapText extends FlxSprite
 		var spaceWidth:Int = font.spaceWidth;
 		var lineLength:Int = line.uLength();
 		var textWidth:Int = this.textWidth;
+		
+		var charUt8;
```
Comment (Gama11) [inline flixel/text/FlxBitmapText.hx:1201]: ?

File: flixel/text/FlxBitmapText.hx [hunk 2]
```diff
@@ -362,6 +362,8 @@ class FlxBitmapText extends FlxSprite
 			}
 
 			var clippedFrameRect;
+			var tmpFrameRect:FlxRect = FlxRect.get();
```
Comment (Gama11) [inline flixel/text/FlxBitmapText.hx:365]: You should call `put()` when you're done with it.

File: flixel/text/FlxBitmapText.hx [hunk 3]
```diff
@@ -1512,6 +1550,7 @@ class FlxBitmapText extends FlxSprite
 		var textLen:Int = Std.int(textData.length / 3);
 		var rect = FlxRect.get();
 		var frameVisible;
+		var prevAdv:Float = 0;
```
Comment (Gama11) [inline flixel/text/FlxBitmapText.hx:1553]: Unused?

**Review by Gama11 — CHANGES_REQUESTED** [review]
Would be good to have a few examples of this feature in the FlxBitmapText demo.

File: flixel/graphics/frames/FlxBitmapFont.hx [hunk 4]
```diff
@@ -660,4 +660,15 @@ class FlxBitmapFont extends FlxFramesCollection
 		font.updateSourceHeight();
 		return font;
 	}
+	
+	/**
+	 * Checks if the specified code is one of the Unicode Combining Diacritical Marks
+	 * @param	Code	The charactercode we want to check
+	 * @return 	Bool	Returns true if the code is a Unicode Combining Diacritical Mark
+	 */
+	public static function isUnicodeComboMark(Code:Int):Bool
```
Comment (Gama11) [inline flixel/graphics/frames/FlxBitmapFont.hx:669]: Not quite sure why you put this here. Do you expect this to be useful on its own as an API that people might use directly? If so, `FlxBitmapFont` is a bit of an awkward place for it, `FlxStringUtil` might be a better fit. If not, I'd just move it to `FlxBitmapText` (the only place it's used) and keep it private for now.

File: flixel/text/FlxBitmapText.hx [hunk 5]
```diff
@@ -445,7 +446,7 @@ class FlxBitmapText extends FlxSprite
 				{
 					dataPos = j * 3;
 
-					currFrame = font.getCharFrame(Std.int(textDrawData[dataPos]));
+					currFrame = font.getCharFrame(Std.int(textDrawData[dataPos])).copyTo();
```
Comment (Gama11) [inline flixel/text/FlxBitmapText.hx:449]: Why is this `copyTo()` necessary now?

File: flixel/text/FlxBitmapText.hx [hunk 6]
```diff
@@ -1208,32 +1209,40 @@ class FlxBitmapText extends FlxSprite
 
 		var tabWidth:Int = spaceWidth * numSpacesInTab;
 
+		var charUt8:UnicodeBuffer;
```
Comment (Gama11) [inline flixel/text/FlxBitmapText.hx:1212]: Why did you move this var declaration? The scope of vars should be kept as small as possible IMO.

**Review by Gama11 — CHANGES_REQUESTED** [review]

File: flixel/graphics/frames/FlxBitmapFont.hx (hunk 4 above)
Comment (SeiferTim) [reply flixel/graphics/frames/FlxBitmapFont.hx:669]: I guess I was thinking that it belonged within the font itself... less than the BitmapText object...
I can move it.

File: flixel/text/FlxBitmapText.hx (hunk 5 above)
Comment (SeiferTim) [reply flixel/text/FlxBitmapText.hx]: Nope.
